src/bika/lims/browser/worksheet/views/referencesamples.py

```diff
@@ -250,6 +250,11 @@ def folderitem(self, obj, item, index):
         # Supported Services
         supported_services_choices = self.make_supported_services_choices(obj)
         item["choices"]["SupportedServices"] = supported_services_choices
+        selected = []
+        for choice in supported_services_choices:
+            if choice.get("selected", False):
+                selected.append(choice)
```

Comment:
You are adding the whole choice (dict) instead of the choice's `ResultValue`. 

This snippet was a suggestion. It will look better if you encapsulate the logic into its own function (e.g. `get_selected_values(self, choices)`), so you can then end up with something easier to read:

```python
item["SupportedServices"] = self.get_selected_values(supported_services_choices)
```